pkg/api/server/v1alpha2/log/s3.go

```diff
@@ -0,0 +1,149 @@
+package log
+
+import (
+	"bufio"
+	"errors"
+	"path/filepath"
+	"strings"
+
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/tektoncd/results/pkg/apis/v1alpha2"
+	"github.com/tektoncd/results/pkg/conf"
+)
+
+type S3LogStreamer struct {
+	LogStreamer
+	io.WriterTo
+	io.ReaderFrom
+
+	bufSize int
+	conf    *conf.ConfigFile
+	path string
+}
+
+func NewS3LogStreamer(trl *v1alpha2.TaskRunLog, bufSize int, conf *conf.ConfigFile, logDataDir string) LogStreamer {
+	if trl.Status.File == nil {
+		trl.Status.File = &v1alpha2.FileLogTypeStatus{
+			Path: defaultFilePath(trl),
+		}
+	}
```

Comment:
This should be an error condition, IMO. The s3 log streamer should not fall back to file as a default.